### PR TITLE
add header supporting google ca regional routing 

### PR DESCRIPTION
**Pull request by quanjielin**

### CODE REVIEW

File: security/pkg/nodeagent/caclient/providers/google/client.go [hunk 1]
```diff
@@ -85,7 +88,16 @@ func (cl *googleCAClient) CSRSign(ctx context.Context, csrPEM []byte, token stri
 		token = bearerTokenPrefix + token
 	}
 
-	ctx = metadata.NewOutgoingContext(ctx, metadata.Pairs("Authorization", token))
+	out, _ := metadata.FromOutgoingContext(ctx)
+	out = out.Copy()
```
Comment (JimmyCYJ) [inline security/pkg/nodeagent/caclient/providers/google/client.go]: Could you add a comment explaining why Copy() is needed here (preventing races by modification)?

File: security/pkg/nodeagent/caclient/providers/google/client.go [hunk 2]
```diff
@@ -110,3 +122,14 @@ func (cl *googleCAClient) getTLSDialOption() (grpc.DialOption, error) {
 	creds := credentials.NewClientTLSFromCert(pool, "")
 	return grpc.WithTransportCredentials(creds), nil
 }
+
+func parseZone(clusterURL string) string {
+	// input: https://container.googleapis.com/v1/projects/quanlinpilot1/locations/us-central1-c/clusters/mpibetacluster2
+	// output: quanlinpilot1 us-central1-c
```
Comment (JimmyCYJ) [inline security/pkg/nodeagent/caclient/providers/google/client.go]: Is this output comment correct? IIUC, FindStringSubmatch() returns ["https://container.googleapis.com/v1/projects/quanlinpilot1/locations/us-central1-c/clusters/mpibetacluster2", "quanlinpilot1", "us-central1-c"]

File: security/pkg/nodeagent/caclient/providers/google/client.go [hunk 3]
```diff
@@ -110,3 +122,14 @@ func (cl *googleCAClient) getTLSDialOption() (grpc.DialOption, error) {
...
+	var rgx = regexp.MustCompile(`.*/projects/(.*)/locations/(.*)/clusters/.*`)
+	rs := rgx.FindStringSubmatch(clusterURL)
+	if len(rs) < 3 {
+		return ""
```
Comment (JimmyCYJ) [inline security/pkg/nodeagent/caclient/providers/google/client.go]: What would happen if return "", does the client send CSR without zone specification or does the client skip sending CSR?

File: security/pkg/nodeagent/caclient/providers/google/client.go (hunk 2 above)
Comment (myidpt) [reply security/pkg/nodeagent/caclient/providers/google/client.go]: Anonymize `quanlinpilot1` to some general name?

File: security/pkg/nodeagent/caclient/providers/google/client.go (hunk 3 above)
Comment (quanjielin) [reply security/pkg/nodeagent/caclient/providers/google/client.go]: without zone info.

**Pull request merged**